test/server/script-processor-test.js

```diff
@@ -356,115 +356,177 @@ describe('Script processor', function () {
 
     it('Should process eval()', function () {
         testProcessing([
-            { src: 'eval(script)', expected: 'eval(__proc$Script(script))' },
-            { src: 'eval("script")', expected: 'eval(__proc$Script("script"))' },
-            { src: 'window.eval(script)', expected: 'window.eval(__proc$Script(script))' },
-            { src: 'window["eval"](script)', expected: 'window["eval"](__proc$Script(script))' },
-
-            { src: 'eval.call(window, script)', expected: 'eval.call(window, __proc$Script(script))' },
-            { src: 'eval.call(window, "script")', expected: 'eval.call(window, __proc$Script("script"))' },
-            { src: 'window.eval.call(window, script)', expected: 'window.eval.call(window, __proc$Script(script))' },
+            {
+                src:      'eval(script)',
+                expected: '(eval === window.eval ? function(script, arg) { return window.eval(__proc$Script(script), arg); } : eval)(script)'
```

Comment:
`setTimeout` and `setInterval` can accept variable number of arguments (https://developer.mozilla.org/en-US/docs/Web/API/WindowTimers/setInterval) and they don't have scope issues like `eval`, so seems like we should keep old transform for timers, and introduce new transform only for eval.